examples/optimization.py

```diff
@@ -14,23 +11,27 @@
 tf_phi = tf.Variable(0.1)
 
 alpha, phi = circuit.params('alpha', 'phi')
+
 with circuit.context as q:
     Dgate(alpha, phi) | q[0]
-results = eng.run(circuit, args={'alpha': tf_alpha, 'phi': tf_phi}, eval=False})
 
-# loss is probability for the coherent state |alpha=1>
-prob = results.state.fidelity_coherent([1.0])
-loss = -prob  # negative sign to maximize prob
+opt = tf.keras.optimizers.Adam(learning_rate=0.1)
+steps = 50
+
+for step in range(steps):
 
-# Set up optimization
-optimizer = tf.train.GradientDescentOptimizer(learning_rate=0.1)
-minimize_op = optimizer.minimize(loss)
+    # reset the engine if it has already been executed
+    if eng.run_progs:
+        eng.reset()
 
-# Create Tensorflow Session and initialize variables
-sess = tf.Session()
-sess.run(tf.global_variables_initializer())
+    with tf.GradientTape() as tape:
+        # execute the engine
+        results = eng.run(circuit, args={'alpha': tf_alpha, 'phi': tf_phi})
```

Comment:
Was `eval=False` removed since that's the default here?

Reply:
Removed since it no longer exists 🙂 In TF2, there is no concept of unevaluated tensors, all tensors are evaluated.